### PR TITLE
Make get zip file task idempotent

### PR DESCRIPTION
**Pull request by amickan**

### CODE REVIEW

File: app/grandchallenge/github/tasks.py
Comment (MikeOverkamp-diag) [inline app/grandchallenge/github/tasks.py:62]: I think this code should not run if the status is `STARTED` as well

**Review by MikeOverkamp-diag — CHANGES_REQUESTED** [review]

File: app/grandchallenge/github/tasks.py
Comment (amickan) [reply app/grandchallenge/github/tasks.py]: True, thanks for catching this!

**Review by MikeOverkamp-diag — APPROVED** [review]
LGTM, but I'd prefer if @jmsmkn had a look at this as well before merging

**Review by jmsmkn — COMMENTED** [review]
I think this looks fine. Generally: get_zipfile does too much. It would be better to split this up into several methods that do only one thing each, then it would be easier to handle unexpected errors with something like:

```
try:
  install_lfs(...)
  fetch_repo(...)
  check_license(...)
  save_zipfile(...)
  build_repo(...)
except: # Broad exception OK here
  # Handle the error and set the failed status
  raise
```

File: app/grandchallenge/github/tasks.py
Comment (amickan) [reply app/grandchallenge/github/tasks.py:65]: I guess you're right. Mike and I had discussed that in some cases we might want to be able to retry, but as it is implementd now we would keep retrying and retrying, which is probably not what we want indeed. @MikeOverkamp-diag ?

Comment (jmsmkn) [reply app/grandchallenge/github/tasks.py:65]: Exactly, if you want to retry then have a retry state and set it to that before running and check for that too (see the implementation for component jobs).

Comment (MikeOverkamp-diag) [reply app/grandchallenge/github/tasks.py:65]: I agree. I think failures in this task generally will not be fixed by a simple retry, so retrying on failure wil only consume unnecessary resources. Maybe we could add a retry mechanism as some point, but I think that's out of scope for this PR.

**Review by MikeOverkamp-diag — APPROVED** [review]

**Pull request merged**
